src/pages/RiskAnalysisEserviceAssociatedPage/RiskAnalysisEServiceAssociated.page.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  PageContainer,
+  SectionContainer,
+  SectionContainerSkeleton,
+} from '@/components/layout/containers'
+import { Box } from '@mui/material'
+import React, { useEffect } from 'react'
+import { Redirect, useParams } from '@/router'
+import { useTranslation } from 'react-i18next'
+import { useSuspenseQuery } from '@tanstack/react-query'
+import { EServiceQueries } from '@/api/eservice'
+import type { EServiceRiskAnalysis } from '@/api/api.generatedTypes'
+import { useSearchParams } from 'react-router-dom'
+import { FormProvider, useForm } from 'react-hook-form'
+import { RHFAutocompleteSingle } from '@/components/shared/react-hook-form-inputs'
+import { RiskAnalysisInfoSummary } from '@/components/shared/RiskAnalysisInfoSummary'
+
+const RiskAnalysisEServiceAssociatedPage: React.FC = () => {
+  return (
+    <React.Suspense fallback={<RiskAnalysisEServiceAssociatedPageContentSkeleton />}>
+      <RiskAnalysisEServiceAssociatedPageContent />
+    </React.Suspense>
+  )
+}
+
+const RiskAnalysisEServiceAssociatedPageContent: React.FC = () => {
+  const { t } = useTranslation('eservice')
+  const { eserviceId, descriptorId } = useParams<'WATCH_RISK_ANALISIS_FOR_ESERVICE'>()
+
+  const { data: descriptor } = useSuspenseQuery(
+    EServiceQueries.getDescriptorProvider(eserviceId, descriptorId)
+  )
+  const [searchParams, setSearchParams] = useSearchParams()
+
+  const riskAnalysisList = descriptor?.eservice.riskAnalysis
+
+  const urlRiskAnalysisId = searchParams.get('riskAnalysisId')
+  const defaultRiskAnalysisId =
+    urlRiskAnalysisId &&
+    riskAnalysisList.some((riskAnalysis) => riskAnalysis.id === urlRiskAnalysisId)
+      ? urlRiskAnalysisId
+      : riskAnalysisList[0].id
+
+  const formMethods = useForm<{ riskAnalysisId: string }>({
+    defaultValues: {
+      riskAnalysisId: defaultRiskAnalysisId,
+    },
+  })
+
+  const riskAnalysisId = formMethods.watch('riskAnalysisId')
+
+  useEffect(() => {
+    if (riskAnalysisId) {
+      setSearchParams((prev) => {
+        prev.set('riskAnalysisId', riskAnalysisId)
+        return prev
+      })
+    } else {
+      setSearchParams((prev) => {
+        prev.delete('riskAnalysisId')
+        return prev
+      })
+    }
+  }, [riskAnalysisId, setSearchParams])
```

Comment:
There are better ways to subscribe to a `react-hook-form` internal state changes. Look [here](https://react-hook-form.com/docs/useform/watch)